minifi-commons/minifi-commons-schema/src/main/java/org/apache/nifi/minifi/commons/schema/BaseSchema.java

```diff
@@ -15,13 +15,17 @@
  * limitations under the License.
  */
 
-package org.apache.nifi.minifi.bootstrap.util.schema.common;
+package org.apache.nifi.minifi.commons.schema;
 
 import java.lang.reflect.Constructor;
 import java.lang.reflect.InvocationTargetException;
+import java.util.Collections;
+import java.util.LinkedHashMap;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 public class BaseSchema {
```

Comment:
So I agree that the toMap method is a little light atm, the one thing I liked about defining it in the superclass was to centralize the decision to use a LinkedHashMap (to preserve insertion order and allow YAML to have attributes in the desired order).

I figured that it was possible we would want to add common attributes at the top level at some point in which case that would be a good place for it.

In terms of the helper methods, I don't have a problem with making them static.